db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/ValueRow.java

```diff
@@ -623,6 +623,21 @@ public StructType createStructType() {
 		}
 		return DataTypes.createStructType(fields);
 	}
+
+	@Override // SPLICE-1621
+	public StructType createStructTypeCorrected(int[] baseColumnMap) {
+		StructField[] fields = new StructField[length()];
+		int j = 0;
+		for(int i = 0; i < baseColumnMap.length; i++){
+			if (baseColumnMap[i] != -1){
+				// put the selected columns, designated in baseColumnMap, in field
+				fields[j] = getColumn(j+1).getStructField(getNamedColumn(i));
+				j++;
+			}
```

Comment:
what happens to the remaining array for fields if the numcols mismatch between baseColumnMap and fields[], should it to get initialized to -1 or something similar.